.github/pull_request_template.md

```diff
@@ -0,0 +1,35 @@
+#### What does the PR do?
+_Describe your pull request here. Please read the text below the line, and make sure you follow the checklist._
+
+
+#### Checklist
+- [ ] Populated all fields in this template.
+- [ ] The PR title reflects the change.
+- [ ] Changes are described in the pull request, or an [existing issue is referenced](https://github.com/triton-inference-server/server/issues).
+- [ ] Added [test plan](#test-plan) and verified test passes.
+- [ ] Verified that the PR passes existing CI.
+- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+#### Type:
+_remove any that do not apply_
+
+- Feature
+- Bugfix
+- Enhancement
+- Chore / doc / build etc
+
+#### Where should the reviewer start?
```

Comment:
Actually should have a place for PR group if the change spans multiple PRs

Reply:
Added as a separate section